engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/NewGameScreen.java

```diff
@@ -15,24 +15,296 @@
 import org.terasology.rendering.nui.CoreScreenLayer;
 import org.terasology.rendering.nui.WidgetUtil;
+import org.terasology.rendering.nui.animation.MenuAnimationSystems;
+import org.terasology.rendering.nui.databinding.Binding;
+import org.terasology.rendering.nui.databinding.ReadOnlyBinding;
+import org.terasology.rendering.nui.itemRendering.StringTextRenderer;
+import org.terasology.rendering.nui.layers.mainMenu.savedGames.GameInfo;
+import org.terasology.rendering.nui.layers.mainMenu.savedGames.GameProvider;
+import org.terasology.rendering.nui.layers.mainMenu.selectModulesScreen.AdvancedGameSetupScreen;
+import org.terasology.rendering.nui.widgets.UIDropdown;
+import org.terasology.rendering.nui.widgets.UIDropdownScrollable;
+import org.terasology.rendering.nui.widgets.UILabel;
+import org.terasology.rendering.nui.widgets.UIText;
+import org.terasology.world.generator.internal.WorldGeneratorInfo;
+import org.terasology.world.generator.internal.WorldGeneratorManager;
+import org.terasology.world.internal.WorldInfo;
+import org.terasology.world.time.WorldTime;
+
+import java.util.Collections;
+import java.util.List;
 
 public class NewGameScreen extends CoreScreenLayer {
 
     public static final ResourceUrn ASSET_URI = new ResourceUrn("engine:newGameScreen");
+    private static final String DEFAULT_GAME_NAME_PREFIX = "Game ";
+    private static final Logger logger = LoggerFactory.getLogger(CreateGameScreen.class);
+    private static final String DEFAULT_GAME_TEMPLATE_NAME = "JoshariasSurvival";
+    private static final String DEFAULT_WORLD_GENERATOR = "Core:FacetedPerlin";
+    private boolean loadingAsServer;
+
+    @In
+    private ModuleManager moduleManager;
+
+    @In
+    private Config config;
+
+    @In
+    private WorldGeneratorManager worldGeneratorManager;
+
+    @In
+    private GameEngine gameEngine;
+
+    @In
+    private TranslationSystem translationSystem;
 
     @Override
-    public void initialise(){
+    public void initialise() {
+
+        setAnimationSystem(MenuAnimationSystems.createDefaultSwipeAnimation());
+
+        UILabel gameTypeTitle = find("gameTypeTitle", UILabel.class);
+        if (gameTypeTitle != null) {
+            gameTypeTitle.bindText(new ReadOnlyBinding<String>() {
+                @Override
+                public String get() {
+                    if (loadingAsServer) {
+                        return translationSystem.translate("${engine:menu#select-multiplayer-game-sub-title}");
+                    } else {
+                        return translationSystem.translate("${engine:menu#select-singleplayer-game-sub-title}");
+                    }
+                }
+            });
+        }
+        final UIText gameName = find("gameName",UIText.class);
+        setGameName(gameName);
+
+        final UIDropdownScrollable<Module> gameplay = find("gameplay", UIDropdownScrollable.class);
+        gameplay.setOptions(getGameplayModules());
+        gameplay.setVisibleOptions(3);
```

Comment:
With the UI a bit more spacious now could it perhaps make sense to bump this to 5?